tests/quantization/test_torchao.py

```diff
@@ -0,0 +1,82 @@
+# SPDX-License-Identifier: Apache-2.0
+"""Test model set-up and inference for TorchAO quantized HF models supported
+ on the CPU/GPU backend.
+
+ Validating the configuration and printing results for manual checking.
+
+ Run `pytest tests/quantization/test_torchao_quant.py`.
+"""
+import pytest
+from torchao.core.config import to_reconstructable_dict
+from torchao.quantization.quant_api import (
+    AffineQuantizedTensor, Float8DynamicActivationFloat8WeightConfig,
+    Int4WeightOnlyConfig, Int8DynamicActivationInt8WeightConfig, PerRow)
+
+MODELS = ["TinyLlama/TinyLlama_v1.1"]
+DTYPE = ["bfloat16"]
+CONFIGS = [
+    Int4WeightOnlyConfig(),
+    Int8DynamicActivationInt8WeightConfig(),
+    Float8DynamicActivationFloat8WeightConfig(granularity=PerRow()),
+]
+
+
+def check_all_weights_quantized(model):
+    """
+    Check if all major linear components in a LLaMA model have weights that are 
+    instances of AffineQuantizedTensor
+    """
+    for layer_idx, layer in enumerate(model.model.layers):
+        # Key linear components to check
+        components = {
+            f"layer{layer_idx}.self_attn.qkv_proj": layer.self_attn.qkv_proj,
+            f"layer{layer_idx}.self_attn.o_proj": layer.self_attn.o_proj,
+            f"layer{layer_idx}.mlp.gate_up_proj": layer.mlp.gate_up_proj,
+            f"layer{layer_idx}.mlp.down_proj": layer.mlp.down_proj
+        }
+
+        for name, component in components.items():
+            # Check if weight is an AffineQuantizedTensor
+            if not hasattr(component, 'weight'):
+                assert isinstance(
+                    component,
+                    AffineQuantizedTensor), f"{name} is not quantized"
+
+
+@pytest.mark.parametrize("model", MODELS)
+@pytest.mark.parametrize("dtype", DTYPE)
+@pytest.mark.parametrize("config", CONFIGS)
+def test_torchao(vllm_runner, model, dtype, config):
+    hf_override_dict = {
```

Comment:
@mgoin Still some work to be done on this PR but here is the hf_overrdide path? 

I do think its kind of nice to be able to quantize on the fly but maybe that isn't as important for most users? If so could be worth revisiting maybe after this PR for better UX

Reply:
I see, thanks for writing these examples. 

This does seem clunky, and I would like it to be easy for users to specify runtime quants if wanted. 

When I proposed the hf_override path, my thought was we don't need to keep the usage so literal to what would need to be in a proper config.json. I think it would be okay if we used the arg to sneak in some special tags/shortcuts, like the torchao_config had in the original PR. If we can interpret those tags in the right place, we can expand them into proper quant_configs. I'll try thinking a bit more tomorrow, let me know if you have ideas